base/applications/mspaint/mouse.cpp

```diff
@@ -58,216 +61,444 @@ BOOL nearlyEqualPoints(INT x0, INT y0, INT x1, INT y1)
     return (abs(x1 - x0) <= cxThreshold) && (abs(y1 - y0) <= cyThreshold);
 }
 
-POINT pointStack[256];
-short pointSP;
+void ToolBase::reset()
+{
+    pointSP = 0;
+}
 
-void
-startPaintingL(HDC hdc, LONG x, LONG y, COLORREF fg, COLORREF bg)
+void ToolBase::OnButtonDown(BOOL bLeftButton, LONG x, LONG y, BOOL bDoubleClick)
```

Comment:
Looking at all Tool implementations, they all _have_ to call the `OnButtonDown` (and most of these functions),
so it might make sense to split it in 2 functions:
1 non-virtual function that handles the required things(updating start/last, whatever), and then calls the 2nd virtual function (that is empty and to be implemented in a tool implementation).

Reply:
OK.